nodes/src/nodes/tool_git/git_repo.py

```
@@ -0,0 +1,952 @@
+        kwargs: Dict[str, Any] = {}
+        if ref:
+            kwargs['newest_commit'] = self._resolve_ref(repo, ref).id
+        try:
+            blame_obj = repo.blame(path, **kwargs)
+        except pygit2.GitError as exc:
+            raise GitError(f'blame failed for {path!r}: {exc}') from exc
+
+        try:
+            entry = repo.head.peel(pygit2.Tree)[path]
+        except KeyError:
+            raise GitError(f'{path!r} not found in HEAD') from None
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)